api/swagger.yml

```diff
@@ -1632,8 +1632,39 @@ components:
       required:
         - installation_id
         - reports
-
-
+    ExternalPrincipalList: 
+      type: object
+      required:
+        - pagination
+        - results
+      properties:
+        pagination:
+          $ref: "#/components/schemas/Pagination"
+        results:
+          type: array
+          items:
+            $ref: "#/components/schemas/ExternalPrincipal"
+    ExternalPrincipalSettings:
+      type: object
+      additionalProperties:
+        type: string
+      description: Additional settings for the external principal.
+    ExternalPrincipal:
+      type: object
+      required:
+        - users
```

Comment:
Why users and not only user IDs?
Where do we use the other information?

Reply:
> Why users and not only user IDs?

Notice `users` is of type `array[string]`, user IDs. 

> Where do we use the other information?

Other information is essentially config that will be digested by the remote authenticator. 
For example, in AWS remote auth -  when we create principal, this will contain flag indicating if we want to use the session name or not. 
It's optional, to stay agile it's not specific to anything so that we don't need to break the API tomorrow.